### DISCUSSION TITLE
Auto Add items to repeater

### DISCUSSION THREAD

**Discussion by ariesdg** [filamentphp/filament#9939] [Help] [answered]
I have a TextInput and a Repeater. I want to search for a product using the TextInput via afterStateUpdated and automatically add the product to the Repeater. I am using the default create page of a resource. Is it possible?

**Comment by leandrocfe** [comment, marked as answer]
```php
TextInput::make('field')
    ->live(onBlur: true)
    ->afterStateUpdated(function (Set $set, Get $get, ?string $state) {

        if (filled($state)) {

            $repeaterItems = $get('repeater');

            //check the product here. If it exists, add it to the array.
            array_push($repeaterItems, ['another_field' => $state]);
            $set('repeater', $repeaterItems);

        }

    }),

Repeater::make('repeater')
    ->schema([
        TextInput::make('another_field'),
    ])
    ->defaultItems(0)
```

**Reply by ariesdg** [reply]
Great, it worked. But how about if I have 1 default item in the repeater?

**Reply by naseef** [reply]
@leandrocfe Thanks. How can I achieve the same functionality if the repeater is a relationship?

**Comment by DeviJv** [comment]
how to auto add when page is loaded bro?

**Reply by leandrocfe** [reply]
`afterStateHydrated()` ?
`default([...])` ?

https://filamentphp.com/docs/3.x/forms/advanced#field-hydration

**Comment by jawadsiddiqui** [comment]
I have payrun filament resource form where I added this repeater below 

```
`
Repeater::make('employeeData')
                    ->relationship()
                    ->schema([
                        
                    ])
                 ->columns(2)`
```

I need this repeater load with all employees mean if I have 10 employee in it show 10 rows in repeater with employee Code thats it

**Reply by presidenwashil** [reply]
i think u can try this


    public static function form(Form $form): Form
    {
        return $form
            ->schema([
                Select::make('grade_id')
                    ->relationship('grade', 'name')
                    ->label('Grade')
                    ->searchable()
                    ->required()
                    ->live(onBlur: false)
                    ->afterStateUpdated(function (Set $set, Get $get, ?string $state) {
                        if (filled($state)) {
                            $repeatItems = $get('attendances') ?? [];

                            $students = Student::where('grade_id', $state)->get();

                            array_push(
                                $repeatItems,
                                ...$students->map(function ($student) {
                                    return [
                                        'student_id' => $student->id ?? null,
                                        'status' => 'present',
                                    ];
                                })->toArray()
                            );

                            $set('attendances', $repeatItems);
                        }
                    }),


                Repeater::make('attendances')
                    ->schema([
                        Select::make('student_id')
                            ->relationship('student', 'name')
                            ->label('Student')
                            ->searchable()
                            ->required(),
                        Select::make('status')
                            ->options([
                                'present' => 'Present',
                                'absent' => 'Absent',
                                'sick' => 'Sick',
                            ])
                            ->label('Status')
                            ->required(),
                    ])
                    ->defaultItems(0)
                    ->minItems(0)
                    ->columns(2)
            ]);
